### ISSUE TITLE
The Android app gets stuck on startup.

### ISSUE THREAD

**Issue by RobertoZaffa** [renpy/renpy#5491]
I built an android app with Renpy and uploaded it to Google Play (.aab file).
Google play console gives me this error:
""This item may not have a label readable by screen readers".
Same error if I load the build of a new VN without modifying it.
I tried downloading the app from Google Play on my phone. When start it remains stuck displaying the presplash file.
The apk file generated by Renpy however works well on my phone.
I extracted the apk from the .aab file with Bundletool and tried to run it with the Android Studio emulator and it crashes like on the phone.
I think at this point there is a problem with how Renpy creates the .aab file.
Amazon instead asks for the APK file and therefore there are no problems: https://www.amazon.com/gp/product/B0D2Y77WP1

I am using:
Renpy 8.2.1.24030407
jdk-21.0.2+13

I attach some screenshots and ask what other information can be used to understand what the problem is.
Thank you
![Screenshot 2024-04-29 232507](https://github.com/renpy/renpy/assets/76481108/fb555889-922b-4c04-9235-1da3a8272cf1)
![Screenshot 2024-04-29 231311](https://github.com/renpy/renpy/assets/76481108/d0bbd751-9355-48b4-a2e3-672e9ab53ea0)

**Comment by renpytom** [comment]
Can you add renpytom@gmail.com to the testers for your app, and sent me your apps name? I'd like to see what errors I get.

**Comment by RobertoZaffa** [comment]
Hi Tom,
first of all, thanks for your help.
I added you as a tester and this is the link: https://play.google.com/apps/internaltest/4701227960855637705
I also attach the logcat generated by Android Studio.
[Pixel-3a-API-34-extension-level-7-x86-64-Android-14_2024-05-01_225046.logcat.txt](https://github.com/renpy/renpy/files/15181312/Pixel-3a-API-34-extension-level-7-x86-64-Android-14_2024-05-01_225046.logcat.txt)

**Comment by RobertoZaffa** [comment]
I did another test.
From Renpy I generated an APK and AAB for "The Question".
I opened the APK with Android Studio and it works.
I extracted the APK from the AAB file with Google Bundle Tool and opened it with Android Studio and when started it crashes like my App (I attach the screenshot).
Consider that this is the first time I've tried to create an App and publish it, I'm not an expert, so I don't rule out the possibility that I'm doing something wrong.
![image](https://github.com/renpy/renpy/assets/76481108/e58c479e-b3bb-43b9-8831-ad8f10ee1bcb)

**Comment by renpytom** [comment]
Extracting the APK won't work - it needs the additonal files that are included in the .aab to work. You need to use bundletool to install it. Ren'Py does that for you.

**Comment by RobertoZaffa** [comment]
Aside from the tests done with Android Studio, have you tried downloading the App from Google Play and running it on an Android device?
The link is: https://play.google.com/apps/internaltest/4701227960855637705 ?

**Comment by RobertoZaffa** [comment]
Just for information, bundletool with the mode=universal option generates a .APKS file which is renamed with a .ZIP extension and once uncompressed two files emerge:
- universal.apk
- toc.pb

toc.pb file is not directly needed to run the APK extracted from the .aab file using bundletool with the --mode=universal option.

Here's why:

Universal APK: The --mode=universal option in bundletool creates a single APK that includes all the resources and code needed for the app to run on any device configuration supported by the app. This eliminates the need for the app store to split the app based on device specifics.

toc.pb for Splitting: The toc.pb file is primarily used when dealing with app bundles containing multiple split APKs. These splits target specific device configurations (like screen size or architecture) to reduce app size.

Universal APK is Self-Contained: Since the universal APK already contains everything, it doesn't need the toc.pb file to understand its own structure. It has all the necessary resources and code within itself to function.
In essence, the toc.pb file becomes irrelevant when using the --mode=universal option. The resulting universal APK is a standalone file with everything it needs to run.

In conclusion, I believe that running universal.apk from Android Studio is the same as downloading the APK from Google Play and running it on an Android device and in fact the behavior is the same, the app crashes as soon as it starts.
I believe there is a problem with how Renpy creates the .AAB file

**Comment by renpytom** [comment]
Does it fail when you use Ren'Py to install the game on a device?

**Comment by RobertoZaffa** [comment]
No, I don't use Ren'Py for installation. 
After downloading the APK onto my phone I start the installation from the phone itself

**Comment by RobertoZaffa** [comment]
Hi Tom,
Do you have any suggestions of what I could try?
Could the Android Studio logcat help you understand what the problem could be?
Have you tried generating an AAB from The Question and then extracting the APK and running it? Or upload the AAB to Google Play and then download the APK and run it? Since the problem is not related to my app but also happens with simple Renpy apps, I think it is a general problem of how AAB files are generated by Renpy.

**Comment by RobertoZaffa** [comment]
Hi Tom,
do you have the opportunity to try an AAB produced with Renpy in Google Play?
Just to understand if it's my problem or a generic Renpy one.
Thank you

**Comment by renpytom** [comment]
You need to use Ren'Py to install the AAB, downloading it yourself won't work. 

People have gotten it working in Google Play, but Google Play has had bugs as well.

**Comment by RobertoZaffa** [comment]
I produce the AAB with Renpy and upload it to Google Play.
How else can I proceed?
Am I doing something wrong?
Is there a procedure/sequence to follow?

**Comment by renpytom** [comment]
That should work. You do need to uninstall the game from your device and reinstall it from google play to have it work.

**Comment by RobertoZaffa** [comment]
I tried but nothing changes.
Even from Android Studio I have the same behavior as the phone.
1) If I open a project using the APK generated by Renpy it works.
2) If I open the APK downloaded from Google Play or extracted from the AAB file generated by Renpy it doesn't work.
A I attach the logcat for both the first and second situations.
[NOT_OK_Pixel-3a-API-34-extension-level-7-x86-64-Android-14_2024-05-09_163700.logcat.txt](https://github.com/renpy/renpy/files/15264218/NOT_OK_Pixel-3a-API-34-extension-level-7-x86-64-Android-14_2024-05-09_163700.logcat.txt)
[OK_Pixel-3a-API-34-extension-level-7-x86-64-Android-14_2024-05-09_165311.logcat.txt](https://github.com/renpy/renpy/files/15264221/OK_Pixel-3a-API-34-extension-level-7-x86-64-Android-14_2024-05-09_165311.logcat.txt)

**Comment by RubenGarcia** [comment]
Tested on renpy-8.2.1-sdk, both issues persist. 
- This item may not have a label readable by screen readers
- The app gets stuck at Downloading assets
If you want to test with a minimal app, you can use
https://github.com/RubenGarcia/RenpyCat/

Currently testing with renpy-8.2.2.24060703+nightly-sdk

**Comment by RubenGarcia** [comment]
Duplicate of
https://github.com/renpy/renpy/issues/5010
probably worth it closing this one.

**Comment by RubenGarcia** [comment]
By the way, renpy-8.2.2.24060703+nightly worked for me, can you confirm it works for you?

**Comment by renpytom** [comment]
Closing as duplicate (and hopefully fixed).
